UnityProject/Assets/Scripts/Objects/Machines/Jukebox.cs

```diff
@@ -1,285 +1,305 @@
+				SoundManager.StopNetworked(soundSpawnToken);
 
-				currentSongTrackIndex++;
-				UpdateGUI();
+			// JESTER
+			// We unload the music from the library, freeing RAM
+			//SoundManager.Instance.UnloadMusic(SoundManager.Instance.MusicLibrary.ElementAt(currentSongTrackIndex).Path);
 
-				if (IsPlaying)
-					Play();
+			currentSongTrackIndex--;
+			UpdateGUIAsync();
 
-				return true;
-			}
-			else
-				return false;
+			if (IsPlaying)
+				Play();
 		}
+	}
 
-		public void VolumeChange(float newVolume)
+	public bool NextSong()
+	{
+		if (currentSongTrackIndex < SoundManager.Instance.MusicLibrary.Count - 1)
 		{
-			Volume = newVolume;
+			if (IsPlaying)
+				SoundManager.StopNetworked(soundSpawnToken);
 
-			AudioSourceParameters audioSourceParameters = new AudioSourceParameters
-			{
-				Volume = newVolume,
-			};
+			// JESTER
+			// We unload the music from the library, freeing RAM
+			// SoundManager.Instance.UnloadMusic(SoundManager.Instance.MusicLibrary.ElementAt(currentSongTrackIndex).Path);
+
+			currentSongTrackIndex++;
+			UpdateGUIAsync();
+
+			if (IsPlaying)
+				Play();
 
-			ChangeAudioSourceParametersMessage.SendToAll(musics[currentSongTrackIndex].name, audioSourceParameters);
+			return true;
 		}
+		else
+			return false;
+	}
+
+	public void VolumeChange(float newVolume)
+	{
+		Volume = newVolume;
 
-		private void OnDamageReceived(DamageInfo damageInfo)
+		AudioSourceParameters audioSourceParameters = new AudioSourceParameters
 		{
-			if (integrity.integrity <= integrity.initialIntegrity / 2)
-			{
-				Stop();
-			}
-		}
+			Volume = newVolume,
+		};
 
-		private void UpdateGUI()
+		ChangeAudioSourceParametersMessage.SendToAll(soundSpawnToken, audioSourceParameters);
+	}
+
+	private void OnDamageReceived(DamageInfo damageInfo)
+	{
+		if (integrity.integrity <= integrity.initialIntegrity / 2)
 		{
-			List<ElementValue> valuesToSend = new List<ElementValue>();
-			valuesToSend.Add(new ElementValue() { Id = "TextTrack", Value = Encoding.UTF8.GetBytes(TrackPosition) });
-			valuesToSend.Add(new ElementValue() { Id = "TextSong", Value = Encoding.UTF8.GetBytes(SongName) });
-			valuesToSend.Add(new ElementValue() { Id = "TextArtist", Value = Encoding.UTF8.GetBytes(Artist) });
-			valuesToSend.Add(new ElementValue() { Id = "ImagePlayStop", Value = Encoding.UTF8.GetBytes(PlayStopButtonPrefabImage) });
-
-			// Update all UI currently opened.
-			TabUpdateMessage.SendToPeepers(gameObject, NetTabType.Jukebox, TabAction.Update, valuesToSend.ToArray());
+			Stop();
 		}
 	}
-}
+
+	private async void UpdateGUIAsync()
+	{
+		List<ElementValue> valuesToSend = new List<ElementValue>();
+		valuesToSend.Add(new ElementValue() { Id = "TextTrack", Value = Encoding.UTF8.GetBytes(TrackPosition) });
+		valuesToSend.Add(new ElementValue() { Id = "TextSong", Value = Encoding.UTF8.GetBytes(await GetSongNameAsync()) });
+		valuesToSend.Add(new ElementValue() { Id = "TextArtist", Value = Encoding.UTF8.GetBytes(await GetArtistNameAsync()) });
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)